dbms/src/Interpreters/Join.h

```diff
@@ -150,6 +151,21 @@ class Join
         RowRefList(const Block * block_, size_t row_num_) : RowRef(block_, row_num_) {}
     };
 
+    /// Map for a time series
+    using ASOFTimeType = UInt32;
```

Comment:
We probably need a setting here: UInt16/32/64. It's OK for the first version.

Reply:
Agreed that this is on the TODO list of stuff to add somewhere down the line as it becomes necessary.